pkg/v3/plugin/ocr3.go

```diff
@@ -70,8 +78,40 @@ func (plugin *ocr3Plugin[RI]) Outcome(outctx ocr3types.OutcomeContext, query typ
 	panic("ocr3 Outcome not implemented")
 }
 
-func (plugin *ocr3Plugin[RI]) Reports(seqNr uint64, outcome ocr3types.Outcome) ([]ocr3types.ReportWithInfo[RI], error) {
-	panic("ocr3 Reports not implemented")
+func (plugin *ocr3Plugin[RI]) Reports(seqNr uint64, raw ocr3types.Outcome) ([]ocr3types.ReportWithInfo[RI], error) {
+	var (
+		reports []ocr3types.ReportWithInfo[RI]
+		outcome ocr2keepersv3.AutomationOutcome
+		err     error
+	)
+
+	if outcome, err = ocr2keepersv3.DecodeAutomationOutcome(raw); err != nil {
+		return nil, err
+	}
+
+	toPerform := []ocr2keepers.CheckResult{}
+	var gasUsed uint64
+
+	for i, result := range outcome.Performable {
+		if gasUsed+result.GasUsed > plugin.ReportGasLimit || len(toPerform) > plugin.MaxUpkeepBatchSize {
+			encoded, encodeErr := plugin.ReportEncoder.Encode(toPerform...)
+			err = errors.Join(err, encodeErr)
+
+			reports = append(reports, ocr3types.ReportWithInfo[RI]{
+				Report: types.Report(encoded),
+			})
+
+			toPerform = []ocr2keepers.CheckResult{}
+			gasUsed = 0
+
+			continue
```

Comment:
i guess this is the expected behavior, we skip L110-111 and start the next report/toPerform

Reply:
This was a mistake of mine. The intent is to add to a report and clear the tracking info but still add the current result to the tracking.